src/hn/ui/CommentThread.ui

```diff
@@ -1,30 +1,28 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!-- Generated with glade 3.22.2 -->
 <interface>
-  <requires lib="gtk+" version="3.20"/>
+  <requires lib="gtk" version="4.0"/>
   <template class="CommentThread">
     <child>
       <object class="GtkOverlay">
-        <child>
+        
+        <property name="child">
           <object class="GtkScrolledWindow">
-            <property name="visible">True</property>
-            <property name="can_focus">True</property>
+            
             <property name="hscrollbar_policy">never</property>
-            <property name="shadow_type">in</property>
-            <child>
+            <property name="child">
               <object class="GtkViewport">
```

Comment:
You probably want to set `scroll-to-focus` to True on all Viewports. If you're using `ScrolledWindow` without a Viewport, you'll want to add a viewport and set that property.